python/python/lance/dataset.py

```diff
@@ -1517,6 +1517,12 @@ def create_index(
         self._ds.create_index(column, index_type, name, replace, kwargs)
         return LanceDataset(self.uri, index_cache_size=index_cache_size)
 
+    def cache_size_bytes(self) -> int:
+        """
+        Return the total size of the index + file metadata caches in bytes.
+        """
+        return self._ds.cache_size_bytes()
+
```

Comment:
I wonder if we should have some sort of session handle instead. It might be too early, but would be nice to have something users could pass around and re-use across datasets.

Reply:
I've added a new session object.  It can only report the size right now, so you can't use it to pass between datasets, but it will help avoid breaking changes in the future when we do add that capability.